src/blobstreamx.cairo

```diff
@@ -0,0 +1,362 @@
+        /// The associated hash
+        fn get_header_hash(self: @ContractState, _height: u64) -> u256 {
+            self.block_height_to_header_hash.read(_height)
+        }
+        fn get_latest_block(self: @ContractState) -> u64 {
+            self.latest_block.read()
+        }
+    }
+
+    #[abi(embed_v0)]
+    impl IBlobstreamXImpl of IBlobstreamX<ContractState> {
+        fn data_commitment_max(self: @ContractState) -> u64 {
+            self.data_commitment_max.read()
+        }
+        fn set_gateway(ref self: ContractState, new_gateway: ContractAddress) {
+            self.ownable.assert_only_owner();
+            self.gateway.write(new_gateway);
+        }
+        fn get_gateway(self: @ContractState) -> ContractAddress {
+            self.gateway.read()
+        }
+
+        fn get_state_proof_nonce(self: @ContractState) -> u64 {
+            self.state_proof_nonce.read()
+        }
+
+        fn get_header_range_id(self: @ContractState) -> u256 {
+            self.header_range_function_id.read()
+        }
+
+        fn set_header_range_id(ref self: ContractState, _function_id: u256) {
+            self.ownable.assert_only_owner();
+            self.header_range_function_id.write(_function_id);
+        }
+
+        fn get_next_header_id(self: @ContractState) -> u256 {
+            self.next_header_function_id.read()
+        }
+
+        fn set_next_header_id(ref self: ContractState, _function_id: u256) {
+            self.ownable.assert_only_owner();
+            self.next_header_function_id.write(_function_id);
+        }
+
+        /// Prove the validity of the header at the target block and a data commitment for the block range [latestBlock, _targetBlock).
+        /// Used to skip from the latest block to the target block.
+        ///
+        /// # Arguments
+        ///
+        /// * `_target_block` - The end block of the header range proof.
+        fn request_header_range(ref self: ContractState, _target_block: u64) {
+            let latest_block = self.get_latest_block();
+            let latest_header = self.block_height_to_header_hash.read(latest_block);
+            assert(latest_header != 0, TendermintXErrors::LatestHeaderNotFound);
+            // A request can be at most data_commitment_max blocks ahead of the latest block.
+            assert(_target_block > latest_block, TendermintXErrors::TargetBlockNotInRange);
+            assert(
+                _target_block - latest_block <= self.data_commitment_max(),
+                TendermintXErrors::TargetBlockNotInRange
+            );
+
+            let mut input = BytesTrait::new_empty();
+            input.append_u64(latest_block);
+            input.append_u256(latest_header);
+            input.append_u64(_target_block);
+
+            let mut entry_calldata = BytesTrait::new_empty();
+            entry_calldata.append_felt252(selector!("commit_header_range"));
+            input.append_u64(latest_block);
+            input.append_u64(_target_block);
```

Comment:
should these be appending to `entry_calldata`? Also it seems like only `_target_block` should be appended?

Reply:
good catch